File: lib/122.best_time_to_buy_and_sell_stock_II.dart
void main() {
  print(maxProfit([7, 1, 5, 3, 6, 4]));
  print(maxProfit([1, 2, 3, 4, 5]));
  print(maxProfit([7, 6, 4, 3, 1]));
  print(maxProfit([1, 2]));
}

int maxProfit(List<int> prices) {
  var price = 0;
  var lastValue = prices.first;

  for (int i = 1; i < prices.length; i++) {
    if (prices[i] > lastValue) {
      price += prices[i] - lastValue;
    }

    lastValue = prices[i];
  }

  return price;
}

File: src/122.best_time_to_buy_and_sell_stock_II.kt
fun main() {
    println(maxProfit2(intArrayOf(7, 1, 5, 3, 6, 4)))
    println(maxProfit2(intArrayOf(1, 2, 3, 4, 5)))
    println(maxProfit2(intArrayOf(7, 6, 4, 3, 1)))
    println(maxProfit2(intArrayOf(1, 2)))
}

fun maxProfit2(prices: IntArray): Int {
    var price = 0
    var last = prices.first()

    for (i in 1..<prices.size) {
        if (prices[i] > last) {
            price += prices[i] - last
        }

        last = prices[i]
    }

    return price
}